webrtc.html

```diff
@@ -10614,127 +10587,124 @@ <h2 id="sec.idp-error-handling">IdP Error Handling</h2>
         both go through these steps.</p>
-        <pre class="example highlight">
-var signalingChannel = new SignalingChannel();
-var configuration = { iceServers: [{ urls: "stuns:stun.example.org" }] };
-var pc;
-var audio = null;
-var audioSendTrack = null;
-var video = null;
-var videoSendTrack = null;
-var started = false;
+        <pre class="example highlight">const signalingChannel = new SignalingChannel();
+const configuration = { iceServers: [{ urls: "stuns:stun.example.org" }] };
+const audio = null;
+const audioSendTrack = null;
+const video = null;
+const videoSendTrack = null;
+const started = false;
+let pc;
 
 // Call warmup() to warm-up ICE, DTLS, and media, but not send media yet.
-function warmup(answerer) {
+async function warmup(answerer) {
     pc = new RTCPeerConnection(configuration);
     if (!answerer) {
-      audio = pc.addTransceiver("audio");
-      video = pc.addTransceiver("video");
+        audio = pc.addTransceiver("audio");
+        video = pc.addTransceiver("video");
     }
 
     // send any ice candidates to the other peer
-    pc.onicecandidate = function (evt) {
+    pc.onicecandidate = (evt) =&gt; {
         signalingChannel.send(JSON.stringify({ candidate: evt.candidate }));
     };
 
     // let the "negotiationneeded" event trigger offer generation
-    pc.onnegotiationneeded = function () {
-        pc.createOffer().then(function (offer) {
-            return pc.setLocalDescription(offer);
-        })
-        .then(function () {
+    pc.onnegotiationneeded = async () =&gt; {
+        try {
+            const offer = await pc.createOffer();
+            await pc.setLocalDescription(offer);
             // send the offer to the other peer
             signalingChannel.send(JSON.stringify({ desc: pc.localDescription }));
-        })
-        .catch(logError);
+        } catch (err) {
+            console.error(err);
+        }
     };
 
     // once remote track arrives, show it in the remote video element
-    pc.ontrack = function (evt) {
-        if (evt.track.kind === "audio") {
-          if (answerer) {
-            audio = evt.transceiver;
-            audio.setDirection("sendrecv");
-            if (started &amp;& audioSendTrack) {
-              audio.sender.replaceTrack(audioSendTrack);
-            }
-          }
-        } else if (evt.track.kind === "video") {
-          if (answerer) {
-            video = evt.transceiver;
-            video.setDirection("sendrecv");
-            if (started &amp;& videoSendTrack) {
-              video.sender.replaceTrack(videoSendTrack);
+    pc.ontrack = async (evt) =&gt; {
+        try {
+            if (evt.track.kind === "audio") {
+                if (answerer) {
+                    audio = evt.transceiver;
+                    audio.setDirection("sendrecv");
+                    if (started &amp;&amp; audioSendTrack) {
+                        await audio.sender.replaceTrack(audioSendTrack);
```

Comment:
Does it make sense to await `replaceTrack` here, which will delay setting `remoteView.srcObject`?

Reply:
thats true if replaceTrack succeed, but what will happen if failed ?
I think handle error when failed without replace srcObject seems better.

Reply:
Alright. I guess it's just an example so no reason to get too optimized.